fem/bilininteg_gradient.cpp

```diff
@@ -0,0 +1,775 @@
+// Copyright (c) 2019, Lawrence Livermore National Security, LLC. Produced at
```

Comment:
While this is not unique to this PR, these kernels are very hard to read without any comments. As someone who will probably have to write specialized kernels in the future, comments outlining the general method for each operator would be greatly appreciated.

Reply:
I have to second this.  Since these files don't have headers we should put detailed, header-level, documentation here before each function.  At a minimum we should have descriptions of the arguments but, as @jamiebramwell suggests, an overview of how the function works would be ideal.